src/test/projects/project16/pom.xml

```diff
@@ -0,0 +1,48 @@
+<project>
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>org.cb.maven.plugins.exec</groupId>
+  <artifactId>project16</artifactId>
+  <version>0.1</version>
+  <packaging>jar</packaging>
+  <name>Maven Exec Plugin</name>
+  <inceptionYear>2005</inceptionYear>
+
+  <licenses>
+    <license>
+      <name>Apache License 2</name>
+      <url>http://www.apache.org/licenses/LICENSE-2.0.txt</url>
+      <distribution>repo</distribution>
+    </license>
+  </licenses>
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>exec-maven-plugin</artifactId>
+        <executions>
+           <execution>
+              <phase>test</phase>
+              <goals>
+                 <goal>java</goal>
+              </goals>
+           </execution>
+        </executions>
+        <configuration>
+          <mainClass>org.codehaus.mojo.exec.Slf4jMain</mainClass>
+          <!-- simulate plugin deps bringing one binding and project deps bringing another one -->
+          <additionalClasspathElements>
+            <additionalClasspathElements>../../../../target/test-classes</additionalClasspathElements> <!-- test setup requires it -->
```

Comment:
<nit_picking> I would prefer something such `@project.basedir@` or whatever use some interpolation </nit_picking>

Reply:
Hmm, me too but it is not ran with it plugin so it is not interpolated this way. Should I impl it in the "test mock" handling?

Reply:
this should do the trick in main pom 
```
          <scriptVariables>
            <mainproject.testOutputDirectory>${project.build.testOutputDirectory}</mainproject.testOutputDirectory>
          </scriptVariables>
```
in src/test/projects/project16/pom.xml
```
<additionalClasspathElements>@mainproject.testOutputDirectory@</additionalClasspathElements>
```
please rebase as I have added some ghactions in master to verify the build 
thanks!

Reply:
@olamy scriptVariables is in invoker plugin right? The test uses the pseudo mock ran with surefire so not sure how it would match? :thinking:

Reply:
arf sorry for the confusion